mysql-test/main/rowid_filter_innodb.test

```diff
@@ -677,4 +679,77 @@ eval EXPLAIN EXTENDED $q;
 
 DROP TABLE t1,t2;
 
+--echo #
+--echo # MDEV-31154: Fatal InnoDB error or assertion `!is_v' failure upon multi-update with indexed virtual column
+--echo #
+
+--echo # Test with auto generated Primary Key
+--echo #
+
+SET @save_optimizer_switch= @@optimizer_switch;
+SET optimizer_switch='rowid_filter=on';
+
+CREATE TABLE t0(a int);
+INSERT INTO t0 SELECT seq FROM seq_1_to_20;
+ANALYZE TABLE t0 PERSISTENT FOR ALL;
+
+CREATE TABLE t1 (
+    a int,
+    b int as (a * 2) VIRTUAL,
+    f char(200), /* Filler */
+    key (b),
+    key (a)
+) engine=innodb;
+
+INSERT INTO t1 (a, f) SELECT seq, seq FROM seq_1_to_1000;
```

Comment:
Do we really need this many rows to reproduce the bug? The test is quite slow:
```
main.rowid_filter_innodb 'innodb'        [ pass ]  17564
```
Before this change, it was almost 2 seconds faster:
```
main.rowid_filter_innodb 'innodb'        [ pass ]  15993
```
This is a test for debug-instrumented builds. If we need many pages for reproducing an issue, we can use `innodb_limit_optimistic_insert_debug`.

Reply:
I tried to reduce it and didn't succeed. All the rowid filter rests use this kind of setup. Also, when I compared the execution time, it was about the same, but different runs have varied in several seconds

Reply:
@dr-m It is not about Innodb and innodb_limit_optimistic_insert_debug won't help. This is required for generating the right optimizer plan.

Reply:
I am keeping test as it is for now. We may improve in future if we could enforce such plan in optimizer with debug switch or something similar.